src/main/java/org/gridsuite/modification/server/repositories/NetworkModificationRepository.java

```diff
@@ -184,12 +184,14 @@ public void deleteModificationGroup(UUID groupUuid) {
     }
 
     @Transactional // To have the find and delete in the same transaction (atomic)
-    public void deleteModifications(UUID groupUuid, Set<UUID> uuids) {
+    public int deleteModifications(UUID groupUuid, Set<UUID> uuids) {
         List<ModificationEntity> modifications = this.modificationRepository.findAllByGroupId(groupUuid)
                 .stream()
                 .filter(m -> uuids.contains(m.getId()))
                 .collect(Collectors.toList());
+        int count = modifications.size();
```

Comment:
Is this really safe ? I mean deleteAll could fails, (it throws IllegalArgumentException by the way if entities is null) but this still return a desync count of elements (the result of findAllByGroupId)... Maybe it's more robust tu use a Try/Catch here and return only modifications.size() if the deleteAll succeed ?

Reply:
if deleteAll throws, we don't return anything, the exception is propagated.

Reply:
Ha you right, my bad ! It's still seems strange to me to return the count returned before the deleteAll call but ok then.